(DOCSP-19240) Java KMIP (#32)
mongodb-university/csfle-guides: java/README.md

     human-readable.

9. Update the code to insert a document with the regular client. What happens?

## KMIP

> **KMIP support in this guide is to be considered "best effort"**.

Reviewer: I suggest clarifying that "best effort" means unstable dependencies. The PR description seems clearer:

> I used the term "best effort" in the README to signify that the KMIP demonstration in the guide uses unsupported and unstable tools (drivers-evergreen-tools and its pykmip implementation).

Perhaps this?

```suggestion
**KMIP support in this guide relies on unstable dependencies**.
```

Author: Agree this is clearer, made the update!

> This guide uses the `direnv` tool to manage environment variables. 

Reviewer 2: comment: I think it would be helpful to include a link to the direnv git repo (which I think tells people to use brew)

Author: After looking at the [direnv git repo](https://github.com/direnv/direnv) and [direnv.net](https://direnv.net/), it seems like direnv.net may be more helpful as it seems to contain the content of the repos readme file without listing out the source code files of direnv. I've added a link to direnv.net, let me know if this is alright or if you disagree with this decision.

Update the `TRUSTSTORE_PASSWORD` environment variable in your `.envrc` file to your global Java TrustStore password.

Run `direnv allow` to add the `TRUSTSTORE_PASSWORD` and `KEYSTORE_PASSWORD` environment variables.

> Run `source ./configure_certs.sh` to download the drivers tools repo, install certificates, and start your KMIP

Reviewer 2: comment: clone may be more intuitive since they may not see anything download (i.e. wget or curl)
```suggestion
Run `source ./configure_certs.sh` to clone the drivers tools repo, install certificates, and start your KMIP
```

Reviewer 2: comment: It may be better to generalize the repo. "The drivers tools repo" is fairly meaningless but oddly specific. Maybe "a tooling repository"?

> server.

Reviewer: I suggest emphasizing that the KMIP server is unstable and for testing only. PyKMIP must not be used in production environments.

Author: I changed this to "start your unstable kmip test server" and added AFAIK Github's [closest approximation to a warning admonition](https://github.com/github/markup/issues/887#issuecomment-418619745) saying not to use the production server.

     human-readable.

9. Update the code to insert a document with the regular client. What happens?

## KMIP

**KMIP support in this guide relies on unstable dependencies**.

This guide uses the [`direnv`](https://direnv.net/) tool to manage environment variables. 

Update the `TRUSTSTORE_PASSWORD` environment variable in your `.envrc` file to your global Java TrustStore password.

Run `direnv allow` to add the `TRUSTSTORE_PASSWORD` and `KEYSTORE_PASSWORD` environment variables.

Run `source ./configure_certs.sh` to configure certificates start your unstable KMIP
test server.

⚠️ | Do not use the provided KMIP server in a production environment.
:---: | :---

Once the server is running, execute the following command:

    mvn compile exec:java -Dexec.mainClass="com.mongodb.csfle.DataEncryptionKeyCreator"

from the `csfle-examples/java` directory to create a data encryption key.

> Once you have created your key, add your key to your `InsertDataWithEncryptedFields` class and run the following command:

Reviewer 3: I'm not clear what this means.  Looking at InsertDataWithEncryptedFields.java, I don't see where to add the key. Can you make it more clear?

Author: Clarified where to add the key value as suggested
